src/server-extension/resolvers/NotificationResolver/index.ts

```diff
@@ -0,0 +1,272 @@
+import 'reflect-metadata'
+import { Resolver, Mutation, UseMiddleware, Ctx, Args } from 'type-graphql'
+import { EntityManager } from 'typeorm'
+import { AccountOnly } from '../middleware'
+import { Context } from '../../check'
+import { withHiddenEntities } from '../../../utils/sql'
+import {
+  Account,
+  OffChainNotification,
+  RuntimeNotification,
+  RuntimeNotificationProcessed,
+} from '../../../model'
+import { NotificationArgs, SetNotificationPreferencesArgs } from './types'
+
+@Resolver()
+export class NotificationResolver {
+  // Set by depenency injection
+  constructor(private em: () => Promise<EntityManager>) { }
+
+  @Mutation(() => Boolean)
+  @UseMiddleware(AccountOnly)
+  async markOffChainNotificationAsRead(
+    @Args() { notificationId }: NotificationArgs,
```

Comment:
we should also consider an option for passing multiple IDs for the "read all" function